zerver/lib/emoji.py

```diff
@@ -30,6 +35,46 @@ def check_valid_emoji(realm, emoji_name):
     # type: (Realm, Text) -> None
     emoji_name_to_emoji_code(realm, emoji_name)
 
+def emoji_code_is_valid(realm, emoji_code, emoji_type):
+    # type: (Realm, Text, Text) -> bool
+    # For a given realm and emoji type, this function determines whether the
+    # current emoji code is valid or not. In case of a deactivated realm emoji,
+    # its emoji code doesn't count towards a valid emoji code.
+    if emoji_type == "realm_emoji":
+        realm_emojis = realm.get_emoji()
+        if emoji_code in realm_emojis and not realm_emojis[emoji_code]["deactivated"]:
+            return True
+    elif emoji_type == "zulip_extra_emoji" and emoji_code == "zulip":
+        return True
+    elif emoji_type == "unicode_emoji" and emoji_code in codepoint_to_name:
+        return True
+    return False
```

Comment:
These functions work, but I ended up refactoring them to spell out the cases a bit more clearly, and also using more lines (which is helpful in causing our automated line coverage checking to ensure we have tests).  Hopefully you agree it's more readable.

I also think we may want to refactor further to basically merge these functions with the `check_` versions (so we can, e.g., give a better error message for deactivated realm emoji, or for "invalid realm emoji", etc.).